src/update-process-env.js

```diff
@@ -36,6 +33,10 @@ function updateProcessEnv (launchEnv) {
     for (let key in envToAssign) {
       if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key)) {
         process.env[key] = envToAssign[key]
+      } else {
+        if (!process.env[key] && envToAssign[key]) {
```

Comment:
So we'll always assign "preserved" environment variables if they're not already defined?